src/Entities/Donation.php

```diff
@@ -724,4 +730,18 @@ public function modifyDataObject( callable $modificationFunction ) {
 		$this->setDataObject( $dataObject );
 	}
 
+	/**
+	 * @param array $data
+	 * @return array
+	 */
+	private function truncateInsanelyLongFields( array $data ) {
```

Comment:
I wouldn't say having a long values of fields must automatically be considered "malicious" or "insane" - if there is a limit set on length of the value, that's fine, just cut everything exceeding the limit.I'd say `truncateTooLongFields` not `truncateInsanelyLongFields`. Also 2 kB is a bit too little to be called insane for my taste, although surely in our case is at least suspicious.